ttnn/cpp/ttnn/operations/moreh/moreh_dot_op/device/moreh_dot_program_factory.cpp

```diff
@@ -0,0 +1,131 @@
+            {CB::c_intermed0, im0_t},
+            {CB::c_intermed1, im1_t},
+        });
+
+    std::vector<uint32_t> reader_compile_time_args = {
+        (std::uint32_t)tt::operations::primary::is_dram(src0_buffer),
+        (std::uint32_t)tt::operations::primary::is_dram(src1_buffer),
+        *reinterpret_cast<uint32_t *>(&scaler)};
+
+    std::vector<uint32_t> writer_compile_time_args = {(std::uint32_t)CB::c_out0, (std::uint32_t)tt::operations::primary::is_dram(dst_buffer)};
+    const auto reader_kernel_file = "ttnn/cpp/ttnn/operations/moreh/moreh_dot_op/device/kernels/reader_moreh_dot.cpp";
+    const auto writer_kernel_file = "ttnn/cpp/ttnn/operations/moreh/moreh_dot_op/device/kernels/writer_moreh_dot.cpp";
+
+    const auto reader_kernel_id = tt::operations::primary::CreateReadKernel(program, reader_kernel_file, core, reader_compile_time_args);
+    const auto writer_kernel_id = tt::operations::primary::CreateWriteKernel(program, writer_kernel_file, core, writer_compile_time_args);
+
+    vector<uint32_t> compute_kernel_args = {};
+    std::map<string, string> compute_defines;
+    compute_defines["REDUCE_OP"] = "PoolType::SUM";
+    compute_defines["REDUCE_DIM"] = "ReduceDim::REDUCE_ROW";
+
+    const uint32_t core_num = 1;
+    const auto compute_kernel_file = "ttnn/cpp/ttnn/deprecated/tt_dnn/op_library/moreh_dot/single_core/kernels/moreh_dot.cpp";
+    const auto compute_kernel_id =
+        tt::operations::primary::CreateComputeKernel(program, compute_kernel_file, {core, core_num, compute_kernel_args}, compute_defines);
+
+    SetRuntimeArgs(
+        program,
+        reader_kernel_id,
+        core,
+        {src0_buffer->address(), src1_buffer->address(), num_tiles, 0, mask_h, mask_w});
+    SetRuntimeArgs(program, compute_kernel_id, core, {num_tiles, 1});
+    SetRuntimeArgs(program, writer_kernel_id, core, {output_tensor.buffer()->address(), 1, 0});
+
+    const std::vector<Tensor> input_tensors = {input_tensor_a, input_tensor_b};
+
+    return {
+        std::move(program),
+        {.unary_reader_kernel_id = reader_kernel_id, .unary_writer_kernel_id = writer_kernel_id}};
+}
+
+void MorehDotOperation::SingleCore::override_runtime_arguments(
+    cached_program_t& cached_program,
+    const operation_attributes_t& operation_attributes,
+    const tensor_args_t& tensor_args,
+    tensor_return_value_t& tensor_return_value) {
+    auto& program = cached_program.program;
+    auto& unary_reader_kernel_id = cached_program.shared_variables.unary_reader_kernel_id;
+    auto& unary_writer_kernel_id = cached_program.shared_variables.unary_writer_kernel_id;
+
+    const auto& input_tensor = tensor_args.input_tensor_a;
+    auto& output_tensor = tensor_return_value;
+
+    auto src_buffer = input_tensor.buffer();
+    auto dst_buffer = output_tensor.buffer();
+
+    {
+        auto& runtime_args = tt::tt_metal::GetRuntimeArgs(program, unary_reader_kernel_id, CoreCoord{0, 0});
+        runtime_args[0] = src_buffer->address();
```

Comment:
There are two input tensors so both of their buffer addresses must be overrided here